pkgs/trebuchet/bin/trebuchet.dart

```diff
@@ -0,0 +1,212 @@
+    await filterRepo(['--path-rename', ':pkgs/$input/']);
+
+    print('Prefix tags');
+    await filterRepo(['--tag-rename', ':$input-']);
+
+    print('Replace issue references in commit messages');
+    await inTempDir((tempDirectory) async {
+      final regexFile = File(p.join(tempDirectory.path, 'expressions.txt'));
+      await regexFile.create();
+      await regexFile.writeAsString('regex:#(\\d)==>dart-lang/$input#\\1');
+      await filterRepo(['--replace-message', regexFile.path]);
+    });
+
+    print('Create branch at target');
+    await runProcess('git', ['checkout', '-b', 'merge-$input-package']);
+
+    print('Add a remote for the local clone of the moving package');
+    await runProcess(
+      'git',
+      ['remote', 'add', '${input}_package', inputPath],
+    );
+    await runProcess('git', ['fetch', '${input}_package']);
+
+    print('Merge branch into monorepo');
+    await runProcess(
+      'git',
+      [
+        'merge',
+        '--allow-unrelated-histories',
+        '${input}_package/$branchName',
+        '-m',
+        'Merge package:$input into shared tool repository'
+      ],
+    );
+
+    if (push) {
+      print('Push to remote');
+      await runProcess(
+        'git',
+        ['push', '--set-upstream', 'origin', 'merge-$input-package'],
+      );
+    }
+
+    print('DONE!');
+    print('''
+Steps left to do:
+
+- Move and fix workflow files
+${push ? '' : '- Run `git push --set-upstream origin merge-$input-package` in the monorepo directory'}
+- Disable squash-only in GitHub settings, and merge with a fast forward merge to the main branch, enable squash-only in GitHub settings.
+- Push tags to github using `git tag --list '$input*' | xargs git push origin`
+- Follow up with a PR adding links to the top-level readme table.
+- Add a commit to https://github.com/dart-lang/$input/ with it's readme pointing to the monorepo.
+- Update the auto-publishing settings on pub.dev/packages/$input.
+- Archive https://github.com/dart-lang/$input/.
+''');
+  }
+
+  Future<void> runProcess(
+    String executable,
+    List<String> arguments, {
+    bool inTarget = true,
+  }) async {
+    final workingDirectory = inTarget ? targetPath : inputPath;
+    print('----------');
+    print('Running `$executable $arguments` in $workingDirectory');
+    if (!dryRun) {
+      final processResult = await Process.run(
+        executable,
+        arguments,
+        workingDirectory: workingDirectory,
+      );
+      print('stdout:');
+      print(processResult.stdout);
+      if ((processResult.stderr as String).isNotEmpty) {
+        print('stderr:');
+        print(processResult.stderr);
+      }
+      if (processResult.exitCode != 0) {
+        throw ProcessException(executable, arguments);
+      }
+    } else {
+      print('Not running, as --dry-run is set.');
+    }
+    print('==========');
+  }
+
+  Future<void> filterRepo(List<String> args) async {
+    await runProcess(
+      'python3',
```

Comment:
Can we validate the existence of `python3` before we run any actions that might make changes on disk? Perhaps prefix all the operations w/ just a `runProcess()` that does `python3 --version`?